protocols/jabber/block.c

```diff
@@ -0,0 +1,94 @@
+/***************************************************************************\
+*                                                                           *
+*  BitlBee - An IRC to IM gateway                                           *
+*  Jabber module - Handling of blocking                                     *
+*                                                                           *
+*  Copyright 2021 / <>                                                      *
+*                                                                           *
+*  This program is free software; you can redistribute it and/or modify     *
+*  it under the terms of the GNU General Public License as published by     *
+*  the Free Software Foundation; either version 2 of the License, or        *
+*  (at your option) any later version.                                      *
+*                                                                           *
+*  This program is distributed in the hope that it will be useful,          *
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of           *
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            *
+*  GNU General Public License for more details.                             *
+*                                                                           *
+*  You should have received a copy of the GNU General Public License along  *
+*  with this program; if not, write to the Free Software Foundation, Inc.,  *
+*  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.              *
+*                                                                           *
+\***************************************************************************/
+
+#include "jabber.h"
+
+int jabber_block_feature(struct im_connection *ic)
+{
+	struct jabber_data *jd = ic->proto_data;
+	GSList *features = jd->features;
+	int foundbk = FALSE;
+	struct prpl *prpl = ic->acc->prpl;
+
+	while (features) {
+		if (!strcmp(features->data, XMLNS_BLOCK)) {
+			foundbk = TRUE;
+		}
+		features = g_slist_next(features);
+	}
```

Comment:
`g_slist_find_custom(jd->features, XMLNS_BLOCK, (GCompareFunc) strcmp)`

(edit: this comment is kinda made obsolete by one i wrote later, keep reading)